### ISSUE TITLE
Atomic updates not working fully with streams inside other streams

### ISSUE THREAD

**Issue by shpuld** [paldepind/flyd#179]
Creating a new issue out this bug that I found when working on #177 

Basically when you have more complicated dependencies that are updated inside other streams, atomic updates break and you get more updates than you would expect.

Issue can be seen in a test case like this:
``` javascript
var result = 0;
stream(1).map(function() {
  stream(0)
    .map(function(val) {
      result += 1;
      return val + 1;
    })
    .map(function(val) {
      result += 1;
      return val + 1;
    });
});
assert.equal(result, 2);
```

The assert fails with the value of 4 (and before the fix for #177 it would fail with the value of 3).

**Comment by nordfjord** [comment]
Ok, I've made a rather large PR (#180) to fix this, It solves both #177 and this issue. Got all the tests running finally, sorry for how long it has taken, I haven't had much free time recently.

**Comment by nordfjord** [comment]
Feel free to take a look at it to see if you spot any problems.

**Comment by shpuld** [comment]
Very cool, thank you for this!

**Comment by paldepind** [comment]
The fix by @nordfjord is now on npm as version 0.2.8 :smile:
